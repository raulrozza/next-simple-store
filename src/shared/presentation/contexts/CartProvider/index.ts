export { default as useAddToCart } from './hooks/useAddToCart';
