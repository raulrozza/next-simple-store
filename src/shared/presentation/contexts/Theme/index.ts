export { default as useTheme } from './hooks/useTheme';
export { default as Theme } from './implementations/StyledComponentTheme';
