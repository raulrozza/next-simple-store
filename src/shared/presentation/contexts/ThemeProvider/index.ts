export { default as useTheme } from './hooks/useTheme';
export { default as ThemeProvider } from './implementations/StyledComponentTheme';
