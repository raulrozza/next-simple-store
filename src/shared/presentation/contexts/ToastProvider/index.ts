export { useToastProvider } from './context';
export { default as ToastProvider } from './implementations/ReactToastifyProvider';
