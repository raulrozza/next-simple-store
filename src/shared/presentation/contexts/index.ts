export * from './CartProvider';
export * from './ThemeProvider';
