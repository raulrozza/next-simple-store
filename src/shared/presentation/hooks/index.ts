import { trpc } from '@/shared/infra/http/trpc';

const { useQuery, useMutation } = trpc;

export { useQuery, useMutation };
