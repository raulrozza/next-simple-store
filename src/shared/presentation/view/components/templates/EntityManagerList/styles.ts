import styled, { css } from 'styled-components';

import { MAX_SCREEN_WIDTH } from '@/shared/presentation/constants/screen';
import { Card } from '@/shared/presentation/view/components/atoms';

export const Container = styled.section`
    display: flex;
    justify-content: center;
`;

export const Content = styled.main`
    ${({ theme }) => css`
        width: 100%;
        max-width: ${theme.layout.size(MAX_SCREEN_WIDTH)};
        margin: ${theme.layout.size(1)};
    `}
`;

export const ButtonPanel = styled.div`
    display: flex;
`;

export const List = styled(Card)`
    overflow: hidden;
`;
