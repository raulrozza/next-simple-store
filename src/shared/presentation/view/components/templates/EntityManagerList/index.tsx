import { FC, PropsWithChildren, ReactElement } from 'react';

import Link from 'next/link';

import { Button, Spacing } from '@/shared/presentation/view/components/atoms';

import { ButtonPanel, Container, Content, List } from './styles';

interface EntityManagerListProps<T> {
    addButton: {
        href: string;
        text: string;
    };
    title: string;
    emptyText: string;
    items: T[];
    renderItem: FC<{ item: T }>;
}

function EntityManagerList<T extends { id: string }>({
    addButton,
    title,
    items,
    emptyText,
    renderItem: RenderItem,
}: PropsWithChildren<EntityManagerListProps<T>>): ReactElement {
    return (
        <Container>
            <Content>
                <ButtonPanel>
                    <Link href={addButton.href} passHref>
                        <Button icon="plus" variant="primary" asAnchor>
                            {addButton.text}
                        </Button>
                    </Link>
                </ButtonPanel>

                <Spacing size={2} />

                <h1>{title}</h1>

                <Spacing size={1} />

                {!items.length && <p>{emptyText}</p>}

                <List as="ul">
                    {items.map(item => (
                        <RenderItem key={item.id} item={item} />
                    ))}
                </List>
            </Content>
        </Container>
    );
}

export default EntityManagerList;
