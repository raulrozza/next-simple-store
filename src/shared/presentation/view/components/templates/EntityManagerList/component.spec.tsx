import { render, screen } from '@testing-library/react';

import { ThemeProvider } from '@/shared/presentation/contexts';

import EntityManagerList from './index';

describe('EntityManagerList', () => {
    it('should render the component correctly, displaying all its elements and the item list', () => {
        const view = render(
            <EntityManagerList
                addButton={{
                    href: '/add',
                    text: 'Add',
                }}
                emptyText="There are no items yet."
                items={[
                    {
                        id: 'dshf9sdghsd',
                        slogan: 'Slogan 1',
                    },
                    {
                        id: 'dshf9gfgfdgdhsd',
                        slogan: 'Slogan 2',
                    },
                ]}
                renderItem={({ item }) => <div>{item.slogan}</div>}
                title="Title"
            />,
            {
                wrapper: ThemeProvider,
            },
        );

        expect(screen.getByText('Title')).toBeInTheDocument();
        expect(screen.getByText('Add')).toBeInTheDocument();
        expect(
            screen.queryByText('There are no items yet.'),
        ).not.toBeInTheDocument();
        expect(screen.getByText('Slogan 1')).toBeInTheDocument();
        expect(screen.getByText('Slogan 2')).toBeInTheDocument();
        expect(view).toMatchSnapshot();
    });

    it('should display the empty text when there are no items', () => {
        const view = render(
            <EntityManagerList
                addButton={{
                    href: '/add',
                    text: 'Add',
                }}
                emptyText="There are no items yet."
                items={[]}
                renderItem={() => <div>Item</div>}
                title="Title"
            />,
            {
                wrapper: ThemeProvider,
            },
        );

        expect(screen.getByText('There are no items yet.')).toBeInTheDocument();
        expect(view).toMatchSnapshot();
    });
});
