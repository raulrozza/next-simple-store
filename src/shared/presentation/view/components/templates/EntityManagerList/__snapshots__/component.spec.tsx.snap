// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EntityManagerList should display the empty text when there are no items 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c3 {
  padding: 0.5rem 1rem;
  border-radius: 0.5rem;
  -webkit-transition: background-color 0.2s;
  transition: background-color 0.2s;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 0.25rem;
  color: #1E4243;
}

.c3:hover {
  background-color: #CDD5D5;
}

.c6 {
  background-color: #FFFFFF;
  border-radius: 1rem;
}

.c4 {
  margin-top: 0.5rem;
}

.c5 {
  margin-top: 0.25rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  width: 100%;
  max-width: 50rem;
  margin: 1rem;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 0.25rem;
}

.c7 {
  overflow: hidden;
}

<body>
    <div>
      <section
        class="c0"
      >
        <main
          class="c1"
        >
          <div
            class="c2"
          >
            <a
              class="c3"
              href="/add"
            >
              Add
            </a>
          </div>
          <div
            class="c4"
            size="2"
          />
          <h1>
            Title
          </h1>
          <div
            class="c5"
            size="1"
          />
          <p>
            There are no items yet.
          </p>
          <ul
            class="c6 c7"
          />
        </main>
      </section>
    </div>
  </body>,
  "container": .c3 {
  padding: 0.5rem 1rem;
  border-radius: 0.5rem;
  -webkit-transition: background-color 0.2s;
  transition: background-color 0.2s;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 0.25rem;
  color: #1E4243;
}

.c3:hover {
  background-color: #CDD5D5;
}

.c6 {
  background-color: #FFFFFF;
  border-radius: 1rem;
}

.c4 {
  margin-top: 0.5rem;
}

.c5 {
  margin-top: 0.25rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  width: 100%;
  max-width: 50rem;
  margin: 1rem;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 0.25rem;
}

.c7 {
  overflow: hidden;
}

<div>
    <section
      class="c0"
    >
      <main
        class="c1"
      >
        <div
          class="c2"
        >
          <a
            class="c3"
            href="/add"
          >
            Add
          </a>
        </div>
        <div
          class="c4"
          size="2"
        />
        <h1>
          Title
        </h1>
        <div
          class="c5"
          size="1"
        />
        <p>
          There are no items yet.
        </p>
        <ul
          class="c6 c7"
        />
      </main>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`EntityManagerList should render the component correctly, displaying all its elements and the item list 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c3 {
  padding: 0.5rem 1rem;
  border-radius: 0.5rem;
  -webkit-transition: background-color 0.2s;
  transition: background-color 0.2s;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 0.25rem;
  color: #1E4243;
}

.c3:hover {
  background-color: #CDD5D5;
}

.c6 {
  background-color: #FFFFFF;
  border-radius: 1rem;
}

.c4 {
  margin-top: 0.5rem;
}

.c5 {
  margin-top: 0.25rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  width: 100%;
  max-width: 50rem;
  margin: 1rem;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 0.25rem;
}

.c7 {
  overflow: hidden;
}

<body>
    <div>
      <section
        class="c0"
      >
        <main
          class="c1"
        >
          <div
            class="c2"
          >
            <a
              class="c3"
              href="/add"
            >
              Add
            </a>
          </div>
          <div
            class="c4"
            size="2"
          />
          <h1>
            Title
          </h1>
          <div
            class="c5"
            size="1"
          />
          <ul
            class="c6 c7"
          >
            <div>
              Slogan 1
            </div>
            <div>
              Slogan 2
            </div>
          </ul>
        </main>
      </section>
    </div>
  </body>,
  "container": .c3 {
  padding: 0.5rem 1rem;
  border-radius: 0.5rem;
  -webkit-transition: background-color 0.2s;
  transition: background-color 0.2s;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 0.25rem;
  color: #1E4243;
}

.c3:hover {
  background-color: #CDD5D5;
}

.c6 {
  background-color: #FFFFFF;
  border-radius: 1rem;
}

.c4 {
  margin-top: 0.5rem;
}

.c5 {
  margin-top: 0.25rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  width: 100%;
  max-width: 50rem;
  margin: 1rem;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 0.25rem;
}

.c7 {
  overflow: hidden;
}

<div>
    <section
      class="c0"
    >
      <main
        class="c1"
      >
        <div
          class="c2"
        >
          <a
            class="c3"
            href="/add"
          >
            Add
          </a>
        </div>
        <div
          class="c4"
          size="2"
        />
        <h1>
          Title
        </h1>
        <div
          class="c5"
          size="1"
        />
        <ul
          class="c6 c7"
        >
          <div>
            Slogan 1
          </div>
          <div>
            Slogan 2
          </div>
        </ul>
      </main>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
